### PR TITLE
2.8 updated mesa driver from 2.9

### PR DESCRIPTION
**Pull request by ddotldot**

### CODE REVIEW

**Comment by zz912** [comment]
Hello ddotldot,

I would like test your Pullrequest. Can I ask for *.diff file?

Thank you

**Comment by petterreinholdtsen** [comment]
[zz912]
> Hello ddotldot,
>
> I would like test your Pullrequest. Can I ask for *.diff file?

You should be able to generate it from git using this recipe:

  git clone https://github.com/LinuxCNC/linuxcnc
  cd linuxcnc
  git remote add ddotldot https://github.com/ddotldot/linuxcnc/
  git fetch ddotldot
  git checkout 2.8-updated-mesa-driver
  git diff d379e05c0839d315c0d443484b958ed225cf65e8 > 2.8-updated-mesa-driver.patch

-- 
Happy hacking
Petter Reinholdtsen

**Comment by zz912** [comment]
Thanks, I'll try it tonight.

**Comment by ddotldot** [comment]
Hi **zz912**, diff link: https://github.com/LinuxCNC/linuxcnc/pull/1899.diff

**Comment by andypugh** [comment]
> You should be able to generate it from git using this recipe:

There is a much easier way, you just append .diff to the pull URL.  (As the OP just did)

**Comment by hansu** [comment]
Or you can just checkout that branch with that single command:

` git fetch upstream pull/1899/head:1899`

**Comment by zz912** [comment]
I tried this:
`Or you can just checkout that branch with this single command:`

`git fetch upstream pull/1899/head:1899`

but I get:

`zdenek@CNC:~/linuxcnc/linuxcnc$ git fetch upstream pull/1899/head:1899`
`fatal: 'upstream' does not appear to be a git repository`
`fatal: Could not read from remote repository.`

`Please make sure you have the correct access rights`
`and the repository exists.`
`zdenek@CNC:~/linuxcnc/linuxcnc$ sudo git fetch upstream pull/1899/head:1899`
`[sudo] heslo pro zdenek: `
`fatal: 'upstream' does not appear to be a git repository`
`fatal: Could not read from remote repository.`

`Please make sure you have the correct access rights`
`and the repository exists.`

**Comment by zz912** [comment]
Reciepe by petterreinholdtsen works.
Thank you.

**Comment by petterreinholdtsen** [comment]
[zz912]
> I tried this:
> `Or you can just checkout that branch with this single command:`
>
> `git fetch upstream pull/1899/head:1899`

If you did the 'git clone' in my recipe, I suspect you will have to
replace 'upstream' with 'origin' to get this working.  Your remote names
are different from Hans.  His 'upstream' is your 'origin'.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by hansu** [comment]
Yes sorry, I forgot to mention that "upstream" has to match the name of your remote.

**Comment by zz912** [comment]
I did:
`  git clone https://github.com/LinuxCNC/linuxcnc`
`  cd linuxcnc`
`  git remote add ddotldot https://github.com/ddotldot/linuxcnc/`
`  git fetch ddotldot`
`  git checkout 2.8-updated-mesa-driver`

Then I compiled it. It works:
7i96s-2 8

I am not sure, if I have actual branche 2.8 from github.com/LinuxCNC merged with ddotldot or I have github.com/ddotldot/linuxcnc/.

Is there anything I can do to help merge this PR? My test was quick, but I can thoroughly check the 7i96s card if that helps.

Or is there any procedure to check the PR before merging?

**Comment by jethornton** [comment]
That would be fantastic if you can get the 7i96S to work with 2.8.3!

JT

**Comment by andypugh** [comment]
I am very nervous about making such a large change to such an imprtant hardware driver in an incremental stable release.

**Comment by c-morley** [comment]
Andy, I agree with you in principle -  but I have an idea  to run past you.

If we add this to the 2.8 repo now, that is not a release but if people wish to try it out they can.
Then before the next actual release you could assess better if it is too dangerous or not.
If still too risky we could pull it out.

 And I'm only suggesting this process because of the current supply problems for Mesa boards that are supported in 2.8.

**Comment by zz912** [comment]
If there was no problem with the chips for the 7i96 card and the 7i96 card was easily available, I would not support this PR.

I see the advantage of this PR in the possibility of using the 7i96s card in the stable version.

I consider running a parallel port risky. In the past, the standard for LCNC was 5i23, but nowadays it is better to use Ethernet Mesa.

**Comment by ddotldot** [comment]
The main differences between version 2.9 and 2.8 drivers are the following changes:
- Add index/probe capability to stepgen, make corresponding encoder probe pins match.
- sserial: Increasing max number of data registers to 7.
- Add OutM simple output module support
- Add preliminary 7I96S support
- And several more small fixes.

These changes can be potentially dangerous:
- Add index/probe capability to stepgen, make corresponding encoder probe pins match, from **1 Feb 2021**.
- sserial: Increasing max number of data registers to 7, from **18 Feb 2021**.

About 1.5 years passed after these changes, and no errors in their work were found.
If the driver works correctly in version 2.9, then in version 2.8 this driver will work correctly.

**Comment by andypugh** [comment]
"Add index/probe capability to stepgen, make corresponding encoder probe pins match, from 1 Feb 2021." is https://github.com/LinuxCNC/linuxcnc/commit/57550c89f3e6ce03b380642d4b818a5533c6956d authored by PCW. 

sserial: Increasing max number of data registers to 7, from 18 Feb 2021. is https://github.com/LinuxCNC/linuxcnc/commit/4f1861ae365241866aa394d125a88954eae77dc9 authored by me, so I know exactly how intrusive it is. 

These have both been extensively tested in 2.9, I agree, but they have not been extensively tested in the context of 2.8 so there may by unknown interactions with other changes. 

How was this commit created? TBH I think I would be happier with it as a cherry-pick of the 2.9 commits rather than a combined merge commit here. And even then I am not sure that I would be 100% happy. 

Is the sserial data size increase critical for the 7i96 support? I am very much less nervous about adding support for new GTAGs  and board IDs than I am in altering the smart-serial pin handling for all boards.

**Comment by ddotldot** [comment]
> These have both been extensively tested in 2.9, I agree, but they have not been extensively tested in the context of 2.8 so there may by unknown interactions with other changes.

Probably problems can be with pncconf and / or with users hal-configs,
if there have been changes in pin names or parameters in 2.9 driver.
Apart from this, changes in the driver should not affect anything.

> How was this commit created?

Created checksum list for 2.9 driver files, checked 2.8 driver files, files with incorrect checksum have been replaced with 2.9 driver files and re-checked 2.8 driver files: the checksum of the 2.8 driver files exactly matched the checksum of the 2.9 driver files.

> Is the sserial data size increase critical for the 7i96 support?
 
The sserial data size does not depend on the type of card(7i96 or other), it depends only on the fpga firmware (sserial module type).
If card firmware support sserial b module(HM2_GTAG_SMARTSERIALB), data size = 224 bits, if not(HM2_GTAG_SMARTSERIAL) data size = 96 bits.

**Comment by andypugh** [comment]
The motivation to add this driver update to 2.8 appears to be to support the 7i96. 

If we can support the 7i96 without updating the Hostmot2 Smart serial driver then I would be a lot happier. 

What was your motivation for this driver update?

**Comment by ddotldot** [comment]
My motivation to have lcnc 2.8 with an actual driver, which fully supports hostmot2 hardware.
Support for 7i96s card is not needed for me, I make hostmot2 compatible hardware for my needs.

**Comment by jethornton** [comment]
7i96S board, not the 7i96 which is not available at all.

**Comment by jethornton** [comment]
Here's the situation, the only Mesa board that _is_ available is the 
7i96S. Spartan 6 plant has been shut down for a some time now. This 
means no 5i25, 6i25, 7i76E, 7i92, 7i95, 7i96, 7i97 are available. A new 
user can't use the Debian deb for LinuxCNC because it's too old to know 
about the 7i96S board. Their only option is to build LinxuCNC deb from 
scratch and install that like shown here 
<https://gnipsel.com/linuxcnc/index.html> in my instructions. I've had 
several people give up trying to use a 7i96S because it's just too 
complicated for them.

It seems to me the only thing that needs to be cherry picked is the 
hostmo2 directory...

JT

**Comment by jethornton** [comment]
Thank you for clarifying that.

**Comment by jethornton** [comment]
and thank you for that link, I'll try that out in the morning and update 
my instructions if all goes well.

JT

**Comment by ddotldot** [comment]
Hal pins and params changes in 2.9 driver:
del pins:
- encoder.%02d.latch-enable
- encoder.%02d.latch-polarity

new pins:
- encoder.%02d.probe-enable   (only for encoder version = 0x83 or 0x84)
- encoder.%02d.probe-invert   (only for encoder version = 0x83 or 0x84)
- inm.%02d.enc0-reset
- inm.%02d.enc1-reset
- inm.%02d.enc2-reset
- inm.%02d.enc3-reset
- inmux.%02d.enc0-reset
- inmux.%02d.enc1-reset
- inmux.%02d.enc2-reset
- inmux.%02d.enc3-reset
- outm.%02d.out-%02d
- outm.%02d.invert-%02d
- ssr.%02d.invert-%02d
- stepgen.%02d.position-reset
- stepgen.%02d.position-latch (only for stepgen version = 0xC2)
- stepgen.%02d.index-enable   (only for stepgen version = 0xC2)
- stepgen.%02d.probe-enable   (only for stepgen version = 0xC2)
- stepgen.%02d.index-invert   (only for stepgen version = 0xC2)
- stepgen.%02d.probe-invert   (only for stepgen version = 0xC2)

new param:
- stepgen.%02d.swap_step_dir  (only for stepgen version = 0x42 or 0xC2)

changed from param to pin:
- rcpwmgen.rate

**Comment by andypugh** [comment]
I think that _all_ is required is the outm module, but I could be wrong. 

No Mesa cards currently use the sserialb (7 register) firmware. 
And the 7i96S is (In think) probably entirely usable without the stepgen index etc. 

I am prepared to introduce a new GTAG in the stable branch as that is new code that is only excercised by firmware that uses it. 

JT, are you able to check out this experimental branch and see if that makes the 7i96S work? 

https://github.com/LinuxCNC/linuxcnc/tree/outm_for_2.8

**Comment by jethornton** [comment]
I will do that in the morning.

JT

**Comment by jethornton** [comment]
I built 2.8 from that branch and when I run LinuxCNC with the 7i96S I 
get the following error.

***@***.***:~$ halrun
halcmd: loadrt hostmot2
Note: Using POSIX realtime
hm2: loading Mesa HostMot2 driver version 0.15
halcmd: loadrt hm2_eth board_ip=10.10.10.10
hm2_eth: loading Mesa AnyIO HostMot2 ethernet driver version 0.2
hm2_eth: 10.10.10.10: INFO: Hardware address (MAC): 00:60:1b:16:80:07
hm2_eth: Unrecognized ethernet board found: 7I96S -- port names will be 
wrong
hm2_eth: discovered 7I96S
hm2/hm2_7i96.0: Low Level init 0.15
hm2: Firmware contains unknown function (gtag-45)

**Comment by andypugh** [comment]
>  Unrecognized ethernet board found: 7I96S -- port names will be wrong hm2_eth: discovered 7I96S hm2/hm2_7i96.0: Low Level init 0.15 hm2: Firmware contains unknown function (gtag-45)

This sounds more like a base 2.8 version. Did you run the rip-environment script? 

https://github.com/LinuxCNC/linuxcnc/blob/outm_for_2.8/src/hal/drivers/mesa-hostmot2/hostmot2.c#L315 (etc) and 
https://github.com/LinuxCNC/linuxcnc/blob/outm_for_2.8/src/hal/drivers/mesa-hostmot2/hostmot2.h#L126

Mean that the GTAG-45 should be handled.

**Comment by jethornton** [comment]
No, I cloned the repo, checked out 2.8 and built a deb and installed it.

**Comment by jethornton** [comment]
I'll double check after work that I didn't make a mistake with git 
checkout...

JT

**Comment by ddotldot** [comment]
> No, I cloned the repo, checked out 2.8

why not with "outm_for_2.8"?

**Comment by zz912** [comment]
> JT, are you able to check out this experimental branch and see if that makes the 7i96S work?
> 
> https://github.com/LinuxCNC/linuxcnc/tree/outm_for_2.8

It works for me:
`zdenek@CNC:~/linuxcnc/linuxcnc-outm_for_2.8/scripts$ ./linuxcnc`
`LINUXCNC - 2.8.3`
`Machine configuration directory is '/home/zdenek/linuxcnc/configs/7i96s'
`Machine configuration file is '7i96s.ini'`
`Starting LinuxCNC...`
`emc/iotask/ioControl_v2.cc 820: iov2: can't load tool table.`
`Found file(REL): ./7i96s.hal`
`Note: Using POSIX realtime`
`hm2: loading Mesa HostMot2 driver version 0.15`
`hm2_eth: loading Mesa AnyIO HostMot2 ethernet driver version 0.2`
`hm2_eth: 10.10.10.10: INFO: Hardware address (MAC): 00:60:1b:16:80:30`
`hm2_eth: discovered 7I96S`
`hm2/hm2_7i96s.0: Low Level init 0.15`
`hm2/hm2_7i96s.0: Smart Serial Firmware Version 43`
`hm2/hm2_7i96s.0: 51 I/O Pins used:`
`hm2/hm2_7i96s.0:     IO Pin 000 (TB3-01): InM Input Module #0, pin in0 (Input)`
`hm2/hm2_7i96s.0:     IO Pin 001 (TB3-02): InM Input Module #0, pin in1 (Input)`
`hm2/hm2_7i96s.0:   ..........................`

I dont do full test, I only loaded Gmoccapy with 7i96s and check Halshow, if there new pin inm, outm and I tried turn on out-04 and out-05.

**Comment by zz912** [comment]
I made same test as JT:

zdenek@CNC:~/linuxcnc/linuxcnc-outm_for_2.8/scripts$ halrun
halcmd: loadrt hostmot2
Note: Using POSIX realtime
hm2: loading Mesa HostMot2 driver version 0.15
halcmd: loadrt hm2_eth board_ip=10.10.10.10
hm2_eth: loading Mesa AnyIO HostMot2 ethernet driver version 0.2
hm2_eth: 10.10.10.10: INFO: Hardware address (MAC): 00:60:1b:16:80:30
hm2_eth: discovered 7I96S
hm2/hm2_7i96s.0: Low Level init 0.15
hm2/hm2_7i96s.0: Smart Serial Firmware Version 43
hm2/hm2_7i96s.0: 51 I/O Pins used:
hm2/hm2_7i96s.0:     IO Pin 000 (TB3-01): InM Input Module #0, pin in0 (Input)
hm2/hm2_7i96s.0:     IO Pin 001 (TB3-02): InM Input Module #0, pin in1 (Input)
hm2/hm2_7i96s.0:     IO Pin 002 (TB3-03): InM Input Module #0, pin in2 (Input)
hm2/hm2_7i96s.0:     IO Pin 003 (TB3-04): InM Input Module #0, pin in3 (Input)
hm2/hm2_7i96s.0:     IO Pin 004 (TB3-05): InM Input Module #0, pin in4 (Input)
hm2/hm2_7i96s.0:     IO Pin 005 (TB3-06): InM Input Module #0, pin in5 (Input)
hm2/hm2_7i96s.0:     IO Pin 006 (TB3-07): InM Input Module #0, pin in6 (Input)
hm2/hm2_7i96s.0:     IO Pin 007 (TB3-08): InM Input Module #0, pin in7 (Input)
hm2/hm2_7i96s.0:     IO Pin 008 (TB3-09): InM Input Module #0, pin in8 (Input)
hm2/hm2_7i96s.0:     IO Pin 009 (TB3-10): InM Input Module #0, pin in9 (Input)
hm2/hm2_7i96s.0:     IO Pin 010 (TB3-11): InM Input Module #0, pin in10 (Input)
hm2/hm2_7i96s.0:     IO Pin 011 (TB3-13/TB3-14): SSR #0, pin Out-00 (Output)
hm2/hm2_7i96s.0:     IO Pin 012 (TB3-15/TB3-16): SSR #0, pin Out-01 (Output)
hm2/hm2_7i96s.0:     IO Pin 013 (TB3-17/TB3-18): SSR #0, pin Out-02 (Output)
hm2/hm2_7i96s.0:     IO Pin 014 (TB3-19/TB3-20): SSR #0, pin Out-03 (Output)
hm2/hm2_7i96s.0:     IO Pin 015 (TB3-21/TB3-22): OutM Output Module #0, pin Out-04 (Output)
hm2/hm2_7i96s.0:     IO Pin 016 (TB3-23/TB3-24): OutM Output Module #0, pin Out-05 (Output)
hm2/hm2_7i96s.0:     IO Pin 017 (TB1-02/TB1-03): StepGen #0, pin Step (Output)
hm2/hm2_7i96s.0:     IO Pin 018 (TB1-04/TB1-05): StepGen #0, pin Direction (Output)
hm2/hm2_7i96s.0:     IO Pin 019 (TB1-08/TB1-09): StepGen #1, pin Step (Output)
hm2/hm2_7i96s.0:     IO Pin 020 (TB1-10/TB1-11): StepGen #1, pin Direction (Output)
hm2/hm2_7i96s.0:     IO Pin 021 (TB1-14/TB1-15): StepGen #2, pin Step (Output)
hm2/hm2_7i96s.0:     IO Pin 022 (TB1-16/TB1-17): StepGen #2, pin Direction (Output)
hm2/hm2_7i96s.0:     IO Pin 023 (TB1-20/TB1-21): StepGen #3, pin Step (Output)
hm2/hm2_7i96s.0:     IO Pin 024 (TB1-22-TB1-23): StepGen #3, pin Direction (Output)
hm2/hm2_7i96s.0:     IO Pin 025 (TB2-02/TB2-03): StepGen #4, pin Step (Output)
hm2/hm2_7i96s.0:     IO Pin 026 (TB2-04/TB2-05): StepGen #4, pin Direction (Output)
hm2/hm2_7i96s.0:     IO Pin 027 (TB2-07/TB2-08): Encoder #0, pin A (Input)
hm2/hm2_7i96s.0:     IO Pin 028 (TB2-10/TB2-11): Encoder #0, pin B (Input)
hm2/hm2_7i96s.0:     IO Pin 029 (TB2-13/TB2-14): Encoder #0, pin Index (Input)
hm2/hm2_7i96s.0:     IO Pin 030 (TB2-16/TB2-17): IOPort
hm2/hm2_7i96s.0:     IO Pin 031 (TB2-18/TB2-19): IOPort
hm2/hm2_7i96s.0:     IO Pin 032 (internal): IOPort
hm2/hm2_7i96s.0:     IO Pin 033 (internal): SSR #0, pin AC Ref (internal) (Output)
hm2/hm2_7i96s.0:     IO Pin 034 (P1-01): IOPort
hm2/hm2_7i96s.0:     IO Pin 035 (P1-02): IOPort
hm2/hm2_7i96s.0:     IO Pin 036 (P1-03): IOPort
hm2/hm2_7i96s.0:     IO Pin 037 (P1-04): IOPort
hm2/hm2_7i96s.0:     IO Pin 038 (P1-05): IOPort
hm2/hm2_7i96s.0:     IO Pin 039 (P1-06): IOPort
hm2/hm2_7i96s.0:     IO Pin 040 (P1-07): IOPort
hm2/hm2_7i96s.0:     IO Pin 041 (P1-08): IOPort
hm2/hm2_7i96s.0:     IO Pin 042 (P1-09): IOPort
hm2/hm2_7i96s.0:     IO Pin 043 (P1-11): IOPort
hm2/hm2_7i96s.0:     IO Pin 044 (P1-13): IOPort
hm2/hm2_7i96s.0:     IO Pin 045 (P1-15): IOPort
hm2/hm2_7i96s.0:     IO Pin 046 (P1-17): IOPort
hm2/hm2_7i96s.0:     IO Pin 047 (P1-19): IOPort
hm2/hm2_7i96s.0:     IO Pin 048 (P1-21): IOPort
hm2/hm2_7i96s.0:     IO Pin 049 (P1-23): IOPort
hm2/hm2_7i96s.0:     IO Pin 050 (P1-25): IOPort
hm2/hm2_7i96s.0: registered
halcmd:

**Comment by andypugh** [comment]
If we can demonstrate this minimal patch is enough to make the 7i96S work with 2.8 then I am willing to make a 2.8.4 fairly quickly. But I will check on the mailing list first.

**Comment by zz912** [comment]
I would like ask: what is mailing list?

Could you please specify a release date for 2.8.4. ?
The situation with 7i96s is more complicated. Pncconf still needs to be resolved. Everything is ready, but you need to give time to Mr. phillc54. I described everything here:
https://github.com/LinuxCNC/linuxcnc/issues/1927

I would also like to be able to put the Czech language for Gmoccapy in the 2.8 branch. But I'm waiting to see how this PR turns out.
https://github.com/LinuxCNC/linuxcnc/pull/1896
I don't know yet whether it will be approved or rejected.

**Comment by jethornton** [comment]
That's probably the issue...

**Comment by andypugh** [comment]
> Could you please specify a release date for 2.8.4. ? The situation with 7i96s is more complicated. Pncconf still needs to be resolved. Everything is ready, but you need to give time to Mr. phillc54. I described everything here: #1927

Does JT's MesaCT tool work for the 7i96S? If so then that might be an alternative to changing stable code in PnCConf. 

> I would also like to be able to put the Czech language for Gmoccapy in the 2.8 branch.

Unlike code changes, translations can easily be added to the stable branch as there is no risk of breaking anything.

**Comment by jethornton** [comment]
I can confirm that after checking out the correct branch that the 7i96S 
is working

***@***.***:~$ cd outm
***@***.***:~/outm$ git checkout outm_for_2.8
D    tests/tool-info/random-with-startup-tool/simpockets.tbl.orig
Branch 'outm_for_2.8' set up to track remote branch 'outm_for_2.8' from 
'origin'.
Switched to a new branch 'outm_for_2.8'
***@***.***:~/outm$ git branch
   2.8
   master
* outm_for_2.8

build and install deb

***@***.***:~$ halrun
halcmd: loadrt hostmot2
Note: Using POSIX realtime
hm2: loading Mesa HostMot2 driver version 0.15
halcmd: loadrt hm2_eth board_ip=10.10.10.10
hm2_eth: loading Mesa AnyIO HostMot2 ethernet driver version 0.2
hm2_eth: 10.10.10.10: INFO: Hardware address (MAC): 00:60:1b:16:80:07
hm2_eth: discovered 7I96S
hm2/hm2_7i96s.0: Low Level init 0.15
hm2/hm2_7i96s.0: Smart Serial Firmware Version 43
hm2/hm2_7i96s.0: 51 I/O Pins used:

**Comment by jethornton** [comment]
Yes my mesact builds a configuration for the 7i96s and just about every 
other card and smart serial cards.

JT

**Comment by silopolis** [comment]
Le mer. 24 août 2022 à 21:22, andypugh ***@***.***> a écrit :

> Could you please specify a release date for 2.8.4. ? The situation with
> 7i96s is more complicated. Pncconf still needs to be resolved. Everything
> is ready, but you need to give time to Mr. phillc54. I described everything
> here: #1927 <https://github.com/LinuxCNC/linuxcnc/issues/1927>
>
> Does JT's MesaCT tool work for the 7i96S? If so then that might be an
> alternative to changing stable code in PnCConf.
>

MesaCT FTW!

**Comment by c-morley** [comment]
If your going to put the changes for the card, we should put the changes in for pncconf. I believe its already done in master.

**Comment by andypugh** [comment]
> If your going to put the changes for the card, we should put the changes in for pncconf. I believe its already done in master.

I see this as a bridging tactic, to get us to 2.9 with support for a card that can actually be bought. ie a 2.3.4 release in days and 2.9 in weeks.

**Comment by c-morley** [comment]
Make no sense to be able to use s card but not configure it with the in tree program made for that.

2.9 in weeks are you kidding?

**Comment by andypugh** [comment]
> 2.9 in weeks are you kidding?

Well, it might be several weeks, but I would be disappointed if it wasn't this year. 

I agree that it _should_ be handled by the in-tree configuration utility. But then I should _not_ be considering this at all. It is only the unfortunate fact that it is all-but impossible to buy a compatible Mesa card at the moment that is making me even consider stretching the releasing guidelines this far.

**Comment by silopolis** [comment]
Le mer. 24 août 2022 à 23:32, andypugh ***@***.***> a écrit :

> 2.9 in weeks are you kidding?
>
> Well, it might be several weeks, but I would be disappointed if it wasn't
> this year.
>

Many of us would be...

I agree that it *should* be handled by the in-tree configuration utility.
>

Does it and can it manage config of cards other than Mesa's?

**Comment by jethornton** [comment]
What is _it_?

**Comment by petterreinholdtsen** [comment]
[andypugh]
> Unlike code changes, translations can easily be added to the stable
> branch as there is no risk of breaking anything.

Actually it might cause git conflicts on either weblate or in master,
when the stable branch is merged into master.  If someone update
translations both places, a conflict is almost guaranteed.  This is
already mentioned in <URL: https://github.com/LinuxCNC/linuxcnc/pull/1896 >.
Conflits on Weblate might cause translations to get lost.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by andypugh** [comment]
Ref translation conflicts: I don't really care about making conflicts in master and making work for devs. The thing that I want to avoid at all costs is breaking user configs, user hardware and user body parts.

**Comment by andypugh** [comment]
> What is _it_?

In this context, any card supported by the shipped Hostmot2 driver. And specifically in this context the 7i96s.

**Comment by petterreinholdtsen** [comment]
[andypugh]
> Ref translation conflicts: I don't really care about making conflicts
> in master and making work for devs. The thing that I want to avoid at
> all costs is breaking user configs, user hardware and user body
> parts.

Sure.  I was not really commenting on what you care about, just the
statement that it was no risk of breaking anything.  Loosing translated
strings (aka throwing away volunteer work hours) is breaking something
in my book.  It might be worth the risk, of course.  That is no my
decision to make.  Just want to make sure the decision is a well
informed one.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by silopolis** [comment]
Le jeu. 25 août 2022 à 01:02, petterreinholdtsen ***@***.***>
a écrit :

> [andypugh]
> > Ref translation conflicts: I don't really care about making conflicts
> > in master and making work for devs. The thing that I want to avoid at
> > all costs is breaking user configs, user hardware and user body
> > parts.
>
> Sure. I was not really commenting on what you care about, just the
> statement that it was no risk of breaking anything. Loosing translated
> strings (aka throwing away volunteer work hours) is breaking something
> in my book.


It surely is in mine too.
This project needs everything but discouraging contributors of any kind and
tainting its reputation, enough are already taking care of the latter :/

**Comment by c-morley** [comment]
Jt- pncconf is " it'

Don't merge 2.8 into master ..ever fixes that problem.

Andy ... if you risk the hostmot driver then pncconf is not really a big deal. This will be the first time in its history that anyone asked if it was stable.  It is almost always buggy when a new major release is made. Because we get weeks or days notice.

Let's just add pncconf and the hostmot stuff and wait two or three weeks then push out 2.8.4.

Why is there a huge push to get out 2.8.4 and why didn't we talk about this before 2.8.3 went out?

**Comment by andypugh** [comment]
> Why is there a huge push to get out 2.8.4 and why didn't we talk about this before 2.8.3 went out?

I didn't know about this issue when I pushed out 2.8.3. And it was _you_ who asked for 2.8.3[1]. I just took a long time to get to it. 

[1] Wed, 6 Apr, 04:26

**Comment by c-morley** [comment]
I'm sure I did ask for 2.8.3.
I always hope for an announcement of intention so people can make sure things are in order.
In this case I believe there was a review for you on this patch  before you pushed 2.8.3.

I understand you have had some personal things going on and so were busy with life. fair enough.

To be clear, my bitch is not with you per say, just our process as a whole.
Tradition, for a long while, is to put off a release super long and then suddenly release without proper prep time for everyone outside of IRC.
GUI work takes time, being  actually used to test, as there are no unit tests like our motion controller.
So if you say 'i want to release (a major)' I need really a couple months at least for finishing and testing.
I tried to discuss this in the mail list but didn't get much of a response

For minor releases it's much less requirements for time - usually the code has sat in the repo for many months already.
but an announcement on the maillist would be helpful so someone can say - hey wait, how about this..

Now in 2.8 and hostmot case it's an anomaly. But the risk is the hostmot changes not pncconf changes. if pncconf is broken it's only annoying.
but we can mitigate that by adding the changes and waiting to release a small while - weeks.

So 2.8.3 was way too long coming but releasing 2.8.4 in a a week is way to fast (because of the unusual code). 
If our policy was to release any new code every (say) 4 months then devs could anticipate when work needs to be in by and when it would approximately be 'in the public'. So deadline communication wouldn't be quite as important.

To summarize:
milestones and reasonably timed 'heads up' messages.

**Comment by c-morley** [comment]
Sorry Andy, looks like you were an assignee not a reviewer -my mistake.

**Comment by silopolis** [comment]
Le jeu. 25 août 2022 à 03:52, c-morley ***@***.***> a écrit :

> ...
>
> To summarize:
> milestones
>

May be you could comment or even support my #1960 proposal?

**Comment by deanasowa** [comment]
To LCNC dev team, just bought some Nesa hardware that requires the new 2.9 LinuxCNC version, just here to encourage and thank you in advance for this software update, Mesa has been struggling with Chip procurement issues and the new hardware takes advantage of the 2.9 python advantages so hoping it will be available soon.

**Comment by zz912** [comment]
Hello ddotldot and Andy,

I wanted to ask if it would be possible to make a special Live distribution from this branch?

https://www.linuxcnc.org/iso/linuxcnc-2.8.3-buster-7i96s.iso

This distro should not have a repository so it doesn't make a mess.

I can't imagine how much work it takes to create such a distribution. I keep finding people on Facebook who are unhappy about this situation.

**Comment by andypugh** [comment]
The ISO build process depends on getting packages from the LinuxCNC server. 
My plan is to make a 2.8.4 release including the changes in the outm-for-2,8 branch (which includes just enough changes to support the 7i96) 
I have been waiting o find the free time, and for the Czech translation to be merged.

**Comment by zz912** [comment]
It sounds great. :-)

**Comment by silopolis** [comment]
Le jeu. 25 août 2022 à 00:43, andypugh ***@***.***> a écrit :

> What is *it*?
>
> In this context, any card supported by the shipped Hostmot2 driver. And
> specifically in this context the 7i96s.
>

As top posting doesn't help to know... On my side I was referring to
PnCConf.

I was asking because it seems to me that MesaCT has been much more active
than the one of PnCConf for a while, is now more advanced, and is somehow
closer of Mesa... Making me wondering about the opportunity to keep
developing two tools for the same hardware?...

**Comment by andypugh** [comment]
On Wed, 24 Aug 2022 at 14:50, jethornton ***@***.***> wrote:

> No, I cloned the repo, checked out 2.8 and built a deb and installed it.


Ah, that's where you went wrong ;-)

You need to check out "outm_for_2.8"

(Which should really have been called something else, on further
consideration)

-- 
atp

**Comment by jethornton** [comment]
I was asking about this question and the reference to "it".
